biscuit-parser/src/parser.rs

```diff
@@ -515,11 +504,22 @@ fn expr1(i: &str) -> IResult<&str, Expr, Error> {
 }
 
 fn expr2(i: &str) -> IResult<&str, Expr, Error> {
+    let initial_input = i;
     let (i, initial) = expr3(i)?;
 
     let (i, remainder) = many0(tuple((preceded(space0, binary_op_2), expr3)))(i)?;
 
-    Ok((i, fold_exprs(initial, remainder)))
+    if remainder.len() < 2 {
```

Comment:
if we know that remainder either has 0 or 1 element, maybe it should be called onl once instead of with `many0`?

Reply:
yeah, i wanted to be the least disruptive i could, but yeah, being explicit in the parser is probably better